src/luascript.cpp

```diff
@@ -4283,11 +4283,6 @@ int LuaScriptInterface::luaGameReload(lua_State* L)
 {
 	// Game.reload(reloadType)
 	ReloadTypes_t reloadType = getNumber<ReloadTypes_t>(L, 1);
-	if (!reloadType) {
```

Comment:
Default behaviour of Game::reload is to reload everything. Wouldn't it be better to introduce some validation here?

Reply:
I think Game::reload should check whether the reloadType is valid or not by explicitly handling the RELOAD_TYPE_ALL case and showing a warning for the default case of a switch and reporting the error back but since @ranisalt said that Game::reload should default to RELOAD_TYPE_ALL in #1520 I don't think that's the desired behavior. This PR simply addresses the problem of not being able to pass RELOAD_TYPE_ALL to luaGameReload function.